node_package/src/serverRenderReactComponent.ts

```diff
@@ -83,18 +85,39 @@ Function Component.`);
   }
 
   const consoleReplayScript = buildConsoleReplay();
+  const addRenderingErrors = (resultObject: RenderResult, renderingError: RenderingError) => {
+    resultObject.renderingError = { // eslint-disable-line no-param-reassign
+      message: renderingError.message,
+      stack: renderingError.stack,
+    };
+  }
```

Comment:
I put this logic in a function so this logic can be used inside the async function as well as normally.